web/static/js/thesis-editor.js

```diff
@@ -91,11 +145,15 @@ class ThesisEditor extends React.Component {
   }
 
   savePressed () {
-    const page = {slug: this.pathname(), title: this.pageTitle(), description: this.pageDescription()}
+    const page = {
+      slug:           this.state.path,
+      title:          this.state.title,
+      description:    this.state.description,
+      redirect_url:   this.state.redirectURL,
+      template:       this.state.template
+    }
```

Comment:
I'm now pulling this data from `state` rather than reaching into the dom for it. That's better.